### ISSUE TITLE
Graphical glitches and crashes

### ISSUE THREAD

**Issue by neuralcpu** [crystalidea/uninstall-tool#129]
I'm experiencing very strange issues with Uninstall Tool 3.7.2.

When I launch the app, it does populate the list of applications available, however, it is unresponsive. After a few clicks, the application immediately closes.

I've recorded some videos of the issue. 

I also had the same issue with Geek Uninstaller, where the list would disappear and the application would crash.

https://user-images.githubusercontent.com/26924006/236651581-a9873fa8-0dba-46fa-9933-9e991afcfc4f.mp4

https://user-images.githubusercontent.com/26924006/236651584-1e1fcb95-b3f1-44a8-a8dd-58ed1dd228d8.mp4

**Comment by kleuter** [comment]
Unfortunately, I wasn't able to read those dump files (strange).

Do glitches also happen in the Light Mode (turn dark mode off, please).

**Comment by neuralcpu** [comment]
Thanks for replying.

The glitches do happen in light mode too - I've just tested this.

I'm running Windows 11 (22H2 OS Build 22621.1635).

**Comment by neuralcpu** [comment]
Occasionally, I'l get a requester with no text and an "OK" button with, what appears to be, the system font. I'm guessing - and I'm certainly no application developer - that this may be GDI+ resource exhaustion. This is a guess though. Remember seeing such stuff in the past.

![geek-uninstaller](https://user-images.githubusercontent.com/26924006/236701586-4515f754-cb49-45c3-87bb-29c1d16232b1.png)

**Comment by neuralcpu** [comment]
I've checked Process Explorer, and it seems geek64.exe has consumed 9,999 GDI handles. I understand the limit is 10,000, which explains the graphical glitches and subsequent termination.

Uninstall Tool also reports 10,000 GDI handles in Process Explorer.

![uninstall-tool-gdi](https://user-images.githubusercontent.com/26924006/236702149-a717000a-7c38-4329-9ffa-70c9717ed30f.png)

**Comment by kleuter** [comment]
thanks a lot for this great research, I'll try to find the problem and post the results later asap

**Comment by kleuter** [comment]
Please download this test executable of the app:

[UninstallTool.zip](https://github.com/crystalidea/uninstall-tool/files/11433548/UninstallTool.zip)

Please copy it to the app folder, start the app, **press F9**, it will produce the **UninstallTool.exe.log file**

**Comment by neuralcpu** [comment]
Hi,

I ran this and created the log file. It took a while for the problem to occur this time, and it seemed to occur after all the icons were loaded. I suspect the number of ARP entries is tripping up the program :) 

[UninstallTool.exe.zip](https://github.com/crystalidea/uninstall-tool/files/11472828/UninstallTool.exe.zip)

**Comment by neuralcpu** [comment]
Yes, it's a bit strange. I'm using GDIView from NirSoft as well, and that's giving the same result as Process Monitor. 

I think the tools are great by the way. I would purchase Uninstall Tool if it didn't crash on my system :)

**Comment by kleuter** [comment]
Please do right click on the UninstallTool.exe process in GDIView and select HTML Report -> Selected Items, please share the output here. Thank you

**Comment by neuralcpu** [comment]
Hi,

Here's the output from GDIView :)

[GDI Handles Counter.zip](https://github.com/crystalidea/uninstall-tool/files/11502831/GDI.Handles.Counter.zip)

**Comment by neuralcpu** [comment]
Hi! Did you manage to make any progress with this issue?

**Comment by kleuter** [comment]
Should be fixed now.
